src/components/CampaignPhoneNumbersForm.jsx

```diff
@@ -253,66 +334,369 @@ export default class CampaignPhoneNumbersForm extends React.Component {
   areaCodeTable() {
     const { inventoryPhoneNumberCounts: reservedNumbers } = this.formValues();
     const assignedNumberCount = this.getTotalNumberCount(reservedNumbers);
-    const { contactsCount, contactsPerPhoneNumber } = this.props;
+    const {
+      isStarted,
+      inventoryCounts,
+      contactsCount,
+      contactsAreaCodeCounts,
+      contactsPerPhoneNumber
+    } = this.props;
     const numbersNeeded = Math.ceil(contactsCount / contactsPerPhoneNumber);
+    let remaining = numbersNeeded - assignedNumberCount;
 
     const headerColor =
       assignedNumberCount === numbersNeeded
         ? theme.colors.darkBlue
         : theme.colors.red;
 
+    const autoAssignRemaining = () => {
+      let inventory = this.formValues().inventoryPhoneNumberCounts;
+
+      const availableAreaCodes = _.flatten(
+        this.props.phoneNumberCounts.map(phoneNumber => {
+          const foundAllocated = inventory.find(
+            ({ areaCode }) => areaCode === phoneNumber.areaCode
+          ) || { count: 0 };
+
+          /* until we save and navigate back and props.inventoryCounts
+             has values, the phoneNumberCounts will need to have the
+             "form state inventory" subtracted from the available count */
+          const availableCount = !inventoryCounts.length
+            ? phoneNumber.availableCount - foundAllocated.count
+            : phoneNumber.availableCount;
+
+          return Array.from(Array(availableCount)).map(() => ({
+            areaCode: phoneNumber.areaCode,
+            state: phoneNumber.state
+          }));
+        })
+      );
+
+      /* eslint-disable no-param-reassign */
+
+      const matchedFromContacts = _.orderBy(
+        contactsAreaCodeCounts,
+        ["count"],
+        ["desc"]
+      ).reduce((obj, contacts) => {
+        // ignore outlier area codes
+        if (contacts.count < 20 || !remaining) return obj;
+
+        const needed = Math.ceil(contacts.count / 200);
```

Comment:
@codygordon I think this should be `contactsPerPhoneNumber` instead of hardcoded 200

Reply:
correct thank you!